source/MarlinProcessors/src/DrawBeamCalFromDD4hep.cpp

```diff
@@ -352,9 +363,43 @@ void DrawBeamCalFromDD4hep::drawPolarGridRPhi2() {
   c2.cd();
   g.Draw("AXIS");
 
-  for (MapIdVal::iterator it = hitEnergies.begin(); it != hitEnergies.end(); ++it) {
-    unsigned int cellid = it->first;
-    double energy = it->second;
+  MapIdVal hitEDensities;
+  double emin=1.e100;
+  double emax=0;
+  unsigned idmin = 0;
+  unsigned idmax = 0;
+
+  for (auto ehit : hitEnergies) {
+
+    unsigned int cellid = ehit.first;
+    auto const& decoder = *m_seg.segmentation()->decoder();
+    auto llCellID = decoder.toLong(0, cellid);
+    int rBin = decoder.get(llCellID, "r");
+    const double rI = rValues[rBin];
+    const double rO = rValues[rBin+1];
+
+    double area = dd4hep::cm2;
+    if (m_drawDensities) area = (rO*rO-rI*rI)*pValues[rBin]/2 ;
+    double edensity = ehit.second / area;
+
+    if (edensity < emin && rBin == rValues.size()-2) {
```

Comment:
1) Why this comparison with `rValue.size()-2`? Please add a comment
2) This gives a warning (and thus an error, please fix signed-unsigned comparison

Reply:
1. Added the comment.
2. Added static cast to long int for rValues.size()